app/javascript/src/controller_services.js

```diff
@@ -562,68 +614,52 @@ function adjustOverviewHeight($overview) {
  **/
 function applyOverviewScroll($overview) {
   var $container = $("<div></div>");
-  var $children = $overview.children();
-  var scrollTimeout;
+  var $main = $("#main-content");
+  var timeout;
 
-  $container.addClass("FlowOverviewScrollingFrame");
-  $container.height($overview.height()); // first steal the height from overview then reset
-  $overview.height("auto");              // the overview so it is controlled by container.
+  $container.addClass("FlowOverviewScrollFrame");
+  $overview.before($container);
+  $container.append($overview);
 
-  $overview.append($container);
-  $container.append($children);
+  adjustScrollDimensionsAndPosition($container);
 
-  // Apply initial adjustment.
-  adjustOverviewScrollDimensions($overview, $container);
-
-  // Listen for screen changes to reapply.
   $(window).on("resize", function() {
-    scrollTimeout = setTimeout(function() {
-      clearTimeout(scrollTimeout);
-      adjustOverviewScrollDimensions($overview, $container);
-    }, 500);
+    clearTimeout(timeout);
+    $main.css("visibility", "hidden");
+    timeout = setTimeout(function() {
+      $container.get(0).style = ""; // reset everything
+      adjustScrollDimensionsAndPosition($container);
+      $main.css("visibility", "visible");
+    }, 1500);
```

Comment:
1.5s seems quite a high timeout on resize - is there a reason for it being so large?

Reply:
It's an arbitrary number so figure is not based on a calculation. Reason for a slightly larger number is because this can be triggered multiple times by either user rotating a tablet device (although we're not aiming for that audience, yet) or by resizing the browser (e.g. drag the window).

Because human interaction (by intention or accident) is involved, the higher number tries to ensure things have settled down before then going ahead with an attempt to redraw the view.

In other words, the lower the number, the more chance there is of the view trying to redraw while other events are being triggered (screen still being dragged, etc.). Again, no calculation; just guesswork, hope, and experimenting to try and get a "pause... and... refresh" feel over a "redraw! redraw! redraw!!" experience.